zmsapi/src/Zmsapi/ProcessAddLog.php

```diff
@@ -34,16 +34,16 @@ public function readResponse(
 
         $isError = Validator::param('error')->isNumber()->setDefault(0)->getValue();
         if ($isError) {
-            Query::writeLogEntry("MTA failed, message=". $mimepart->content,
-                $processId,
-                Query::PROCESS,
-                $process->scope->id
+            Query::writeProcessLog(
+                "MTA failed, message=" . $mimepart->content,
+                Query::ACTION_MAIL_SUCCESS,
+                $process
             );
         } else {
-            Query::writeLogEntry("MTA successful, subject=". $mimepart->content,
-                $processId,
-                Query::PROCESS,
-                $process->scope->id
+            Query::writeProcessLog(
+                "MTA successful, subject=" . $mimepart->content,
+                Query::ACTION_MAIL_FAIL,
+                $process
```

Comment:
_:warning: Potential issue_

**Fix the logical error in the condition block**

There appears to be a mismatch between the `$isError` condition and the logged actions. Currently, a success action is logged when there's an error, and vice versa. This needs to be corrected to ensure accurate logging.

Please apply the following changes:

```diff
 if ($isError) {
     Query::writeProcessLog(
         "MTA failed, message=" . $mimepart->content,
-        Query::ACTION_MAIL_SUCCESS,
+        Query::ACTION_MAIL_FAIL,
         $process
     );
 } else {
     Query::writeProcessLog(
         "MTA successful, subject=" . $mimepart->content,
-        Query::ACTION_MAIL_FAIL,
+        Query::ACTION_MAIL_SUCCESS,
         $process
     );
 }
```


<!-- suggestion_start -->
<details>
<summary>📝 Committable suggestion</summary>

> ‼️ **IMPORTANT**
> Carefully review the code before committing. Ensure that it accurately replaces the highlighted code, contains no missing lines, and has no issues with indentation. Thoroughly test & benchmark the code to ensure it meets the requirements.

`````suggestion
            Query::writeProcessLog(
                "MTA failed, message=" . $mimepart->content,
                Query::ACTION_MAIL_FAIL,
                $process
            );
        } else {
            Query::writeProcessLog(
                "MTA successful, subject=" . $mimepart->content,
                Query::ACTION_MAIL_SUCCESS,
                $process
`````

</details>
<!-- suggestion_end -->

<!-- This is an auto-generated comment by CodeRabbit -->